tests/unit/classifier/Perceptron_unittest.cc

```diff
@@ -68,16 +72,68 @@ TEST(Perceptron, custom_hyperplane_initialization)
 	auto test_features = wrap(env->get_features_test());
 	auto test_labels = wrap(env->get_labels_test());
 
-	auto perceptron = some<CPerceptron>(features, labels);
-	perceptron->train();
+	auto perceptron = some<CPerceptron>();
+	perceptron->set_labels(labels);
+	perceptron->train(features);
 
 	auto weights = perceptron->get_w();
 
-	auto perceptron_initialized = some<CPerceptron>(features, labels);
+	auto perceptron_initialized = some<CPerceptron>();
 	perceptron_initialized->set_initialize_hyperplane(false);
 	perceptron_initialized->set_w(weights);
-	perceptron_initialized->set_max_iter(1);
+	perceptron_initialized->put<index_t>("max_iteration", 1);
+	perceptron_initialized->set_labels(labels);
 
-	perceptron_initialized->train();
+	perceptron_initialized->train(features);
 	EXPECT_TRUE(perceptron_initialized->get_w().equals(weights));
 }
+
+TEST(Perceptron, prematurely_stop)
```

Comment:
we will need to merge [this](https://github.com/shogun-toolbox/shogun/pull/4327/files#diff-5334f04115d44684a239e44b3ce21fbaL20) before we can run this test otherwise it will just time out in an infinite loop.

Reply:
on it